archive/indexer.go

```diff
@@ -8,21 +8,50 @@ import (
 
 	"github.com/brimsec/zq/zbuf"
 	"github.com/brimsec/zq/zdx"
-	"github.com/brimsec/zq/zio/detector"
+	"github.com/brimsec/zq/zio/bzngio"
 	"github.com/brimsec/zq/zng"
 	"github.com/brimsec/zq/zng/resolver"
 )
 
 const zarExt = ".zar"
 
-// TBD
+// XXX Embedding the type and field names like this can result in some clunky
+// file names. We might want to re-work the naming scheme.
+
+func typeZdxName(t zng.Type) string {
+	return "zdx:type:" + t.String()
+}
+
+func fieldZdxName(fieldname string) string {
+	return "zdx:field:" + fieldname
+}
+
+func archiveDir(path string) (string, error) {
+	//XXX for now the index directory is the name of the zng file
+	// with the ".zar" extension
+	subdir := path + zarExt
+	// make subdirectory for index if it doesn't exist
+	if err := os.Mkdir(subdir, 0755); err != nil {
+		if !os.IsExist(err) {
+			return "", err
+		}
+	}
+	return subdir, nil
+}
+
+// Indexer provides a means to index a zng file.  First, a stream of zng.Records
+// is written to the Indexer via zbuf.Writer, then the indexed records are read
+// as a stream via zbuf.Reader.   The index is managed as a zdx bundle.
```

Comment:
```suggestion
// as a stream via zbuf.Reader.  The index is managed as a zdx bundle.
```

Reply:
I still do 2 spaces after period, but yeah, 3 is too much.

Reply:
I'm all about two spaces between sentences.